client/app/configuration/pick-ami/PickAmiController.js

```diff
@@ -76,16 +78,24 @@ angular.module('EnvironmentManager.configuration').controller('PickAmiController
       if (!$scope.DataLoading && !$scope.InitialVersionSet) {
         var amiVersion = awsService.images.GetAmiVersionFromName(currentAmi);
         var initialAmiVersion = GetAmiByVersion(amiVersion, $scope.SelectedImageType.Versions);
-        initialAmiVersion = initialAmiVersion || $scope.SelectedImageType.Versions[0];
+        initialAmiVersion = initialAmiVersion ||
+          firstStable($scope.SelectedImageType.Versions) ||
+          $scope.SelectedImageType.Versions[0];
         $scope.Version.SelectedVersion = initialAmiVersion;
         $scope.InitialVersionSet = true;
       } else {
         if ($scope.SelectedImageType && $scope.SelectedImageType.Versions) {
-          $scope.Version.SelectedVersion = $scope.SelectedImageType.Versions[0];
+          $scope.Version.SelectedVersion = firstStable($scope.SelectedImageType.Versions) || $scope.SelectedImageType.Versions[0];
```

Comment:
Nice work handling the case when no stable image exists for the selected image type.